godel_plugins/include/godel_plugins/widgets/surface_detection_configuration.h

```diff
@@ -0,0 +1,32 @@
+#ifndef SURFACE_DETECTION_CONFIGURATION_H
+#define SURFACE_DETECTION_CONFIGURATION_H
+
+#include <godel_plugins/widgets/parameter_window_base.h>
+#include <ui_surface_detection_configuration.h>
+#include <godel_msgs/SurfaceDetectionParameters.h>
+
+
+namespace godel_plugins
+{
+
+class SurfaceDetectionConfigWidget : public ParameterWindowBase
+{
+  Q_OBJECT
+public:
+  SurfaceDetectionConfigWidget(godel_msgs::SurfaceDetectionParameters params);
```

Comment:
My personal preference is to pass arguments as `const &` whenever possible as it may help avoid the small computational penalty incurred by the extra copying operation.   In this case, that penalty is probably negligible.